sonic-net/sonic-buildimage
# Changes to support sequential IDF isolation

#### Why I did it
In order to isolate an IDF sequentially (one device at a time), need a mechanism to perform downstream isolation on T2 device. Adding support for two kinds of isolation -  
1. By setting no-export BGP community on all routes advertised to T1 (except loopback), so the routes are not advertised to T0s
2. By withdrawing all routes from T1

##### Work item tracking
- Microsoft ADO **(number only)**: #27100957, #27101099

#### How I did it
Added
1.  New configuration to support the changes described by https://github.com/sonic-net/sonic-buildimage/pull/18597
2. Script to update configDB on each ASIC to configure no-export/withdraw-all isolation and unisolation of IDF. This is copied to host
3. BGPcfgd changes to handle configDB events and add/remove the required route-maps
4. UT cases and data file changes for BGPCfgd and jinja templates

#### How to verify it


Corresponding templates on T2 have to be updated to verify the functionality

Configuration options-
1. sudo idf_isolation withdraw_all
2. sudo idf_isolation no-export
3. sudo idf_isolation unisolated

Validate the routes on T1 are either withdrawn, tagged with no-export community or re-advertised based on the configuration

#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
